crates/sncast/src/starknet_commands/declare.rs

```diff
@@ -75,8 +77,14 @@ pub async fn declare(
 
     let contract_definition: SierraClass = serde_json::from_str(&contract_artifacts.sierra)
         .context("Failed to parse sierra artifact")?;
-    let casm_contract_definition: CompiledClass =
-        serde_json::from_str(&contract_artifacts.casm).context("Failed to parse casm artifact")?;
+    let casm_contract_definition: CompiledClass = if contract_artifacts.casm.is_empty() {
+        let sierra: Value = serde_json::from_str(&contract_artifacts.sierra).unwrap();
+        let casm = serde_json::to_string(&compile(sierra).unwrap()).unwrap();
```

Comment:
Let's maybe avoid unwraping, and try to handle possible error(s) with expect/context ?